src/toil/spark/spark_cluster.py

```diff
@@ -0,0 +1,328 @@
+# Copyright (C) 2016 UCSC Computational Genomics Lab
+# Copyright (C) 2016 UC Berkeley AMPLab
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Spawns a Spark cluster backed by HDFS.
+
+@author Audrey Musselman-Brown, almussel@ucsc.edu
+@author Frank Austin Nothaft, fnothaft@berkeley.edu
+"""
+
+import logging
+import multiprocessing
+import os
+import subprocess
+import sys
+import time
+
+from toil.job import Job
+
+_log = logging.getLogger(__name__)
+_SPARK_MASTER_PORT = "7077"
+
+#
+# FIXME: Where should this go?
+# See discussion in https://github.com/BD2KGenomics/toil-scripts/pull/190.
+# @benedictpaten thinks this should go in toil. If so, where in toil?
+#
```

Comment:
The _docker_call method makes sense. I think you and @jvivian should chat about this and associated docker methods so we can get this as a library module within toil, and not have it duplicated within toil scripts.